src/angular-bootstrap-toggle.js

```diff
@@ -1,289 +1,337 @@
+								2;
+							self.width = wrapperComputedWidth + "px";
+						}
 
-                    // Special case: empty on and off labels (replace with blank space)
-                    if (self.on === '') {
-                      self.on = '&nbsp;';
-                    }
-                    if (self.off === '') {
-                      self.off = '&nbsp;';
-                    }
+						// Calculate the proper height
+						if (!self.height || reevaluate) {
+							var wrapperComputedHeight = Math.max(
+									self.onElement.scrollHeight,
+									self.offElement.scrollHeight) +
+								2;
+							self.height = wrapperComputedHeight + "px";
+						}
+					};
 
-                    this.init = function (ngModelCtrl_) {
-                        ngModelCtrl = ngModelCtrl_;
+					this.init = function(ngModelCtrl_) {
+						ngModelCtrl = ngModelCtrl_;
 
-                        var labels = self.element.find('label');
-                        var spans  = self.element.find('span');
-                        var divs   = self.element.find('div');
+						var labels = self.element.find("label");
+						var spans = self.element.find("span");
+						var divs = self.element.find("div");
 
-                        self.wrapperElement = divs[0];
-                        self.onElement = labels[0];
-                        self.offElement = labels[1];
-                        self.handleElement = spans[0];
+						self.wrapperElement = divs[0];
+						self.onElement = labels[0];
+						self.offElement = labels[1];
+						self.handleElement = spans[0];
 
-                        // Set wigget's visible text such as On/Off or Enable/Disable
-                        angular.element(self.onElement).html(self.on);
-                        angular.element(self.offElement).html(self.off);
+						// Set wigget's visible text such as On/Off or Enable/Disable
+						angular.element(self.onElement).html(self.on);
+						angular.element(self.offElement).html(self.off);
 
-                        // Set the button size
-                        angular.element(self.wrapperElement).addClass(self.size);
-                        angular.element(self.onElement).addClass(self.size);
-                        angular.element(self.offElement).addClass(self.size);
-                        angular.element(self.handleElement).addClass(self.size);
+						// Set the button size
+						angular.element(self.wrapperElement).addClass(self.size);
+						angular.element(self.onElement).addClass(self.size);
+						angular.element(self.offElement).addClass(self.size);
+						angular.element(self.handleElement).addClass(self.size);
 
-                        // Set the toggleClass on the wrapper
-                        angular.element(self.wrapperElement).addClass(self.toggleClass);
+						// Set the toggleClass on the wrapper
+						angular.element(self.wrapperElement).addClass(self.toggleClass);
+						var jElement = $(this.onElement);
```

Comment:
I love the idea of tracking the visibility! Can we make this more AngularJS-ish, like the below?
```js
$scope.$watch(function() { return angular.element(self.wrapperElement).is(':visible') }, function() {
    evaluateSize(true);
});
```
Then no need for jQuery, etc. I haven't tried yet, though... ;-)

Reply:
sure i can wrap the IntersectionObserver into angularjs ... but not today

Reply:
i lied ... i had a few minutes... jquery removed

Reply:
the watch will be invoked every diggest cycle ... its more or less the same as an timeout and i am no friend of that ... i sticked to the IntersectionObserver.